frontend/src/components/HomePage.vue

```diff
@@ -1,46 +1,23 @@
 <template>
   <div>
-    <div class="modal fade" id="modal" tabindex="-1" aria-labelledby="newTravelModalLabel" aria-hidden="true">
-      <div class="modal-dialog modal-dialog-centered modal-lg modal-fullscreen-sm-down">
-        <div class="modal-content">
-          <div class="modal-header">
-            <h5 v-if="modalMode === 'add'" class="modal-title">{{ $t('labels.newX', { X: $t('labels.' + modalObjectType) }) }}</h5>
-            <h5 v-else-if="modalObject" class="modal-title">{{ modalObject.name }}</h5>
-            <button type="button" class="btn-close" @click="hideModal()"></button>
-          </div>
-          <div v-if="modalObject" class="modal-body">
-            <template v-if="modalObjectType === 'travel'">
-              <TravelApplication
-                v-if="modalMode === 'view'"
-                :travel="modalObject as TravelSimple"
-                @cancel="hideModal()"
-                @edit="showModal('edit', modalObject, 'travel')"
-                @deleted="deleteTravel"></TravelApplication>
-              <TravelApplyForm
-                v-else
-                :mode="modalMode"
-                @cancel="hideModal()"
-                :travel="modalObject as Partial<TravelSimple>"
-                @add="applyForTravel"
-                @edit="applyForTravel"
-                ref="travelApplyForm"></TravelApplyForm>
-            </template>
-            <ExpenseReportForm
-              v-else-if="modalObjectType === 'expenseReport'"
-              :mode="modalMode as 'add' | 'edit'"
-              :expenseReport="modalObject as Partial<ExpenseReportSimple>"
-              @cancel="hideModal()"
-              @add="addExpenseReport"></ExpenseReportForm>
-            <HealthCareCostForm
-              v-else
-              :mode="modalMode as 'add' | 'edit'"
-              :healthCareCost="modalObject as Partial<HealthCareCostSimple>"
-              @cancel="hideModal()"
-              @add="addHealthCareCost"></HealthCareCostForm>
-          </div>
-        </div>
+    <ModalComponent @hideModal="hideModal()"
```

Comment:
Warum wird das schließen des Modals in der `HomePage` behandelt?

Auch das erstellen des bootstrap Objects könnte doch besser im `ModalComponent` einmal für alle behandelt werden.
(Meine In Zeile 168 und folgende)
```
 mounted() {
    const modalEL = document.getElementById('modal')
    if (modalEL) {
      this.modal = new Modal(modalEL, {})
    }
  },
```

Reply:
ich habe die hideModal Methode nicht ausgelagert, da sie unter anderem this.modalObject = undefined ausführt. Mir erschien es nicht sinnvoll, den Zugriff auf die Variablen in ModalComponent zu ermöglichen, um das Modal zu schließen. 
Ähnliches dachte ich bei der Instanziierung, zwar wäre es deutlich schöner, aber in anderen Dateien, in denen die modal component verwendet wird, werden beim Anlegen zusätzlich andere Funktionen ausgeführt, sodass mir ein einheitliches Anlegen des Modals nicht sinnvoll erschien. 

Eventuell sehe/kenne ich die Möglichkeiten nicht, wie ich einfacher Variablen und Datenfluss zwischen Parent und Childkomponenten durchführen kann. Generell wäre es natürlich schöner,  wenn die Funktion und das Modal nur einmal angelegt werden und überall verwendet werden können. 

(wie ich auf Funktionen einer Childkomponente in einer Parentmethode zugreiffen kann, weiß ich, der Rest erschließt sich mir allerdings noch nicht so ganz)

Reply:
Ich will dich einladen, die `ModalComponent` mehr losgelöst von den aktuellen Implementierungen der Modals zusehen und mehr als Generische Komponente, die ich an jeder Stelle der App für ein Modal benutzen kann.

Es muss nicht unbedingt ein `modalObject` geben. Was dann in dem Modal angezeigt und gemacht wird kann der Komponente ganz egal sein.

Die Basic Funktionalität eines Modals sollte komplett in der Komponente abgebildet sein, also Instanzierung und auch das Schließen des Modals.

Reply:
Ich habe beides jetzt ausgelagert in die ModalComponente und an vielen stellen benutze ich auch die hideModal Methode aus der ModalComponent, aber um auf die emits von den Komponenten zu reagieren, habe ich immernoch eine hideModal Methode die wiederum die Methode in der ModalComponent aufruft. Das erscheint mir nicht optimal. 

Die "reset" Logik, die bisher nach dem hide() ausgeführt wurde, müsste noch angepasst werden. Ich würde denken, da macht ein allgemeiner emit aus der ModalComponent sinn, der in den Elternkomponenten individuell die Logik behandelt?